cpu/nrf9160/spi_twi_irq.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2021 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_nrf9160
+ * @{
+ *
+ * @file
+ * @brief       Shared IRQ handling between SPI and TWI peripherals on the nRF52
+ *              devices
+ *
+ *              I2C is called TWI (Two Wire Interface) in the datasheets from Nordic
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ *
+ * @}
+ */
+
+#include <assert.h>
+
+#include "cpu.h"
+#include "periph_cpu.h"
+#include "periph_conf.h"
+
+static spi_twi_irq_cb_t _irq[SPIM_COUNT];
+static void *_irq_arg[SPIM_COUNT];
```

Comment:
Isn't `SPIM_COUNT` the number of *SPI* Master interfaces?

Reply:
It is, there is up to 4 SPIM on nRF9160 but there are shared with I2C and UART. Like Atmel SERCOMs.

Reply:
Ok, but better use `TWIM_COUNT` still :wink: